libc/src/__support/UInt.h

```diff
@@ -952,6 +952,35 @@ struct make_signed<UInt<Bits>> : type_identity<Int<Bits>> {
                 "Number of bits in Int should be a multiple of 64.");
 };
 
+namespace internal {
+template <typename T> struct is_custom_uint : cpp::false_type {};
+template <size_t Bits> struct is_custom_uint<UInt<Bits>> : cpp::true_type {};
+} // namespace internal
+
+// bit_cast to UInt
+template <typename To, typename From,
+          typename = cpp::enable_if_t<internal::is_custom_uint<To>::value>,
+          typename = cpp::enable_if_t<sizeof(To) == sizeof(From)>,
+          typename = cpp::enable_if_t<cpp::is_trivially_copyable<From>::value>>
```

Comment:
Why using separate template parameters for the conditions instead of logically combining in just one `enable_if_t`?

Reply:
For consistency with the original implementation
https://github.com/llvm/llvm-project/blob/41096d19ab07650747a434345842a9e55fa972d7/libc/src/__support/CPP/bit.h#L28-L36

Now we could also pack everything in a single condition for both implementations at the _cost_ of explicitly picking up a different type for `std::enable_if` (otherwise they would both resolve to `template <typename To, typename From, void>` and compiler would complain it can't choose between the two).

Reply:
As discussed with @legrosbuffle the list of `typename`s is surprising so I'll change it to a single `enable_if_t` with a conjunction. I'll add a tag for the specialization so the signatures differ.

Reply:
So this was not possible to use a single `typename` for function discrimination. I added an extended explanation. Let me know if it works for you.